src/contracts/access_control/GranularGuardianAccessControl.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.8;
+
+import {ICrossChainForwarder} from '../interfaces/ICrossChainForwarder.sol';
+import {ICrossChainControllerWithEmergencyMode} from '../interfaces/ICrossChainControllerWithEmergencyMode.sol';
+import {IGranularGuardianAccessControl, Envelope, ICrossChainReceiver} from './IGranularGuardianAccessControl.sol';
+import {AccessControlEnumerable} from 'openzeppelin-contracts/contracts/access/AccessControlEnumerable.sol';
+import {IWithGuardian} from 'solidity-utils/contracts/access-control/OwnableWithGuardian.sol';
+
+error DefaultAdminCantBe0();
+error CrossChainControllerCantBe0();
+error NewGuardianCantBe0();
```

Comment:
I would define the errors on the `IGranularGuardianAccessControl` interface instead of here

Reply:
fixed